test/integration/admin/api/services_controller_test.rb

```diff
@@ -3,69 +3,214 @@
+      service = provider.services.last!
+      assert_equal 'published', service.state
+
+      put admin_api_service_path(service, access_token: access_token_value, format: :json), permitted_params.merge({state_event: 'publish'})
+      assert_equal 'published', service.reload.state
+    end
+
+    private
+
+    def assert_correct_params
+      service = provider.services.last!
+      permitted_params.each do |attribute_name, param_value|
+        assert check_equality_value(attribute_name, service.public_send(attribute_name), param_value), "#{attribute_name} does not have the expected value of #{param_value}"
+      end
+      forbidden_params.each do |attribute_name, param_value|
+        refute check_equality_value(attribute_name, service.public_send(attribute_name), param_value), "#{attribute_name} should not have the value of #{param_value}"
+      end
+    end
+
+    def check_equality_value(attribute_name, attribute_value, param_value)
+      if attribute_name == :notification_settings
+        return false if !attribute_value.is_a?(Hash) || (attribute_value.keys.length != param_value.keys.length)
+
+        attribute_value.each do |name_notification_setting, values|
+          return false if values.map(&:to_i) != param_value[name_notification_setting].map(&:to_i)
+        end
+
+        true
+      else
+        param_value == attribute_value
+      end
+    end
+
+    def permitted_params
+      @permitted_params ||= {
+        name: 'the-name',
+        description: 'New description',
+        support_email: 'support@example.com',
+        deployment_option: 'hosted',
+        backend_version: '2',
+        buyers_manage_keys: false,
+        buyer_key_regenerate_enabled: false,
+        mandatory_app_key: false,
+        intentions_required: true,
+        buyers_manage_apps: false,
+        referrer_filters_required: true,
+        custom_keys_enabled: false,
+        buyer_can_select_plan: true,
+        txt_support: 'text for txt support',
+        terms: 'these are our terms',
+        buyer_plan_change_permission: 'direct',
+        notification_settings: {
+          web_provider:   ['', '50', '100', '100'],
+          email_provider: ['', '50', '100', '150'],
+          web_buyer:      ['', '50', '100', '200'],
+          email_buyer:    ['', '50', '100', '300']
+        }
+      }
+    end
+
+    def forbidden_params
+      @forbidden_params ||= {
+        created_at: 1.day.from_now,
+        updated_at: 1.day.from_now,
+        kubernetes_service_link: '/api/v1/namespaces/example-project/services/example-api',
+        account_id: provider.id + 1,
+        tenant_id:  provider.id + 1,
+        oneline_description: 'one line description',
+        txt_api: 'text for txt api',
+        txt_features: 'text for txt features',
+        draft_name: 'example of draft name',
+        infobar: 'infobar text',
```

Comment:
Some are present in a drop or view so I'd suppose it can be modified

```ruby
app/controllers/api/terms_controller.rb
17:      redirect_to edit_admin_service_terms_path(@edited_service)

lib/developer_portal/app/views/developer_portal/settings/terms.html.erb
4:  <%= raw textilize(@service.terms) %>
```

```ruby
lib/developer_portal/lib/liquid/drops/service.rb
198:      def infobar
199:        @service.infobar
```
```ruby
lib/developer_portal/app/views/shared/_footer.html.erb
4:    <% if @service.try(:txt_support).present? %>
```

Reply:
here you are mentioning 3 attributes:
- terms
- txt_support
- infobar

Terms and txt_support are being allowed in the strong params, infobar isn't, but read the comment in the drop:
https://github.com/3scale/porta/blob/ad46a58a31aba8889e419c48da369ae2cfed807e/lib/developer_portal/lib/liquid/drops/service.rb#L196-L200
And read my justification of why I am not allowing it:
https://issues.jboss.org/browse/THREESCALE-2715?focusedCommentId=13769613&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13769613